cantusdb/csv/merge.py

```diff
@@ -0,0 +1,43 @@
+"""
+Merge all raw Cantus CSV into one Dataframe and export to a new CSV
+"""
+
+import os
+import glob
+import pandas as pd
+
+INPUT_PATH = os.path.join(os.path.dirname(__file__), "../data/raw")
+OUTPUT_PATH = os.path.join(os.path.dirname(__file__), "../data/cantus.csv")
+MAPPING_PATH = os.path.join(os.path.dirname(__file__), "../data/mappings")
+
+# Create an empty list to store individual DataFrames
+dfs = []
+genre_mappings = pd.read_csv(
+    os.path.join(MAPPING_PATH, "genres.tsv"), sep="\t", header=0, index_col=False
+)
+service_mappings = pd.read_csv(
+    os.path.join(MAPPING_PATH, "services.tsv"), sep="\t", header=0, index_col=False
+)
+genre_mappings = dict(zip(genre_mappings["Genre"], genre_mappings["Description"]))
+service_mappings = dict(zip(service_mappings["Service"], service_mappings["Description"]))
+
+# Iterate over all files in the directory
+for filename in glob.glob(f"{INPUT_PATH}/*.csv", recursive=False):
+    # Read the CSV file into a DataFrame
+    df = pd.read_csv(filename)
+    entity_id = (filename.split("/")[-1])[0:6] # get the 6-digit source id
```

Comment:
Sources can have IDs longer than 6 characters (see, for example https://cantusdatabase.org/source/1000019)

Reply:
I will try another approach.